src/shared/sam/debugger/goSamDebug.ts

```diff
@@ -84,8 +88,7 @@ export async function makeGoConfig(config: SamLaunchRequestArgs): Promise<GoDebu
     const port: number = config.debugPort ?? -1
 
     config.codeRoot = pathutil.normalize(config.codeRoot)
-    config.debuggerPath = path.join(tempDirPath, 'godbg')
-
+    config.debuggerPath = path.join(os.tmpdir(), 'aws-toolkit-vscode', 'godbg')
```

Comment:
any reason not to use `makeTemporaryToolkitFolder()` ?

Reply:
Docker didn't work too well with Delve being in '/tmp' (only for MacOS though...)
Now that I think about it, it would be better to store the Go debug stuff in VSCode's 'globalStoragePath'.

Reply:
Note that the globalStoragePath only works if this doesn't have to be mounted in the Docker container (e.g. C# downloading the debugger to the project dir, which is then mounted and run inside the container).

Reply:
Yeah it does need to be mounted. I'll leave Delve in a temporary folder for now so at least the binary will persist across multiple debug sessions and extension reloads.

Reply:
In that case, should we store delve in the local workspace (instead of a temporary folder)? That would be consistent with how we manage vsdbg for C#/dotnet. https://github.com/aws/aws-toolkit-vscode/blob/8863ba4a799ee1b465411b8019b547b242a42293/src/shared/sam/debugger/csharpSamDebug.ts#L207

Also wonder if this "download and install to workspace" logic can be abstracted, i.e. is there much opportunity for reuse between the C# and Go cases?